src/Generator/Generators/CLI/CLITypePrinter.cs

```diff
@@ -146,11 +146,11 @@ public override TypePrinterResult VisitDelegate(FunctionType function)
             Enumeration @enum;
             if (pointee.TryGetEnum(out @enum))
             {
-                var typeName = @enum.Visit(this);
+                var typeName = VisitDeclaration(@enum, quals);
 
                 // Skip one indirection if passed by reference
-                if (Parameter != null && (Parameter.IsOut || Parameter.IsInOut)
-                    && pointee == finalPointee)
+                if (Parameter != null && (Parameter.Type.IsReference() || ((Parameter.IsOut || Parameter.IsInOut)
```

Comment:
This line is now too long, you could break it at `((Parameter.IsOut`.

Reply:
Done